pkg/gatewayconfigurationserver/middleware.go

```diff
@@ -15,39 +15,60 @@
 package gatewayconfigurationserver
 
 import (
-	echo "github.com/labstack/echo/v4"
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
 	"go.thethings.network/lorawan-stack/v3/pkg/auth/rights"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/webhandlers"
+	"go.thethings.network/lorawan-stack/v3/pkg/webmiddleware"
+	"google.golang.org/grpc/metadata"
 )
 
-const gatewayIDKey = "gateway_id"
+func (s *Server) fillRequestContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := s.FillContext(r.Context())
```

Comment:
The `FillContext` middleware is already on the root router, so this shouldn't be necessary, right?